pkg/user/appointee/can_call.go

```diff
@@ -0,0 +1,153 @@
+			return canCall(c)
+		},
+		After: telemetry.AfterRunAction(),
+		Flags: []cli.Flag{
+			&AccountAddressFlag,
+			&CallerAddressFlag,
+			&TargetAddressFlag,
+			&SelectorFlag,
+			&PermissionControllerAddressFlag,
+			&flags.NetworkFlag,
+			&flags.EnvironmentFlag,
+			&flags.ETHRpcUrlFlag,
+		},
+	}
+
+	return canCallCmd
+}
+
+func canCall(cliCtx *cli.Context) error {
+	ctx := cliCtx.Context
+	logger := common.GetLogger(cliCtx)
+
+	config, err := readAndValidateUserConfig(cliCtx, logger)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to read and validate user can call config", err)
+	}
+
+	elReader, err := getEigenLayerReader(cliCtx, logger, config)
+	if err != nil {
+		return err
+	}
+
+	var result bool
+	result, err = elReader.UserCanCall(ctx, config.UserAddress, config.CallerAddress, config.Target, config.Selector)
+	logger.Infof("CanCall Result: %v", result)
+	logger.Infof("Selector, Target and User: %s, %x, %s", config.Target, string(config.Selector[:]), config.UserAddress)
+	return err
+}
+
+func readAndValidateUserConfig(cliContext *cli.Context, logger logging.Logger) (*CanCallConfig, error) {
+	userAddress := gethcommon.HexToAddress(cliContext.String(AccountAddressFlag.Name))
+	callerAddress := gethcommon.HexToAddress(cliContext.String(CallerAddressFlag.Name))
+	ethRpcUrl := cliContext.String(flags.ETHRpcUrlFlag.Name)
+	network := cliContext.String(flags.NetworkFlag.Name)
+	environment := cliContext.String(flags.EnvironmentFlag.Name)
+	target := gethcommon.HexToAddress(cliContext.String(TargetAddressFlag.Name))
+	selector := cliContext.String(SelectorFlag.Name)
+	selectorBytes, err := common.ValidateAndConvertSelectorString(selector)
+	if err != nil {
+		return nil, err
+	}
+
+	if environment == "" {
+		environment = common.GetEnvFromNetwork(network)
+	}
+
+	chainID := utils.NetworkNameToChainId(network)
+	permissionManagerAddress := cliContext.String(PermissionControllerAddressFlag.Name)
+
+	if common.IsEmptyString(permissionManagerAddress) {
+		permissionManagerAddress, err = common.GetPermissionManagerAddress(utils.NetworkNameToChainId(network))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	logger.Debugf(
+		"Env: %s, network: %s, chain ID: %s, PermissionManager address: %s",
+		environment,
+		network,
+		chainID,
+		permissionManagerAddress,
+	)
+
+	return &CanCallConfig{
+		Network:                  network,
+		RPCUrl:                   ethRpcUrl,
+		UserAddress:              userAddress,
+		CallerAddress:            callerAddress,
+		Target:                   target,
+		Selector:                 selectorBytes,
+		PermissionManagerAddress: gethcommon.HexToAddress(permissionManagerAddress),
+		ChainID:                  chainID,
+		Environment:              environment,
+	}, nil
+}
+
+func getEigenLayerReader(
+	cliContext *cli.Context,
+	logger logging.Logger,
+	config *CanCallConfig,
+) (UserCanCallReader, error) {
+	if reader, ok := cliContext.App.Metadata["elReader"].(UserCanCallReader); ok {
```

Comment:
what is this? why is reader in cli context?

Reply:
It's a common dependency injection strategy. Providing implementations by injection or generating if not provided. It's used in the unit test.

Reply:
i am not convinced this is the right way of doing it. like having the reader and context and casting it seems ugly. And according to the code the `ok` will never be true for actual command. so that means we are adding a path in actual command just for unit test. I am not too sure about. I will think over it tho.
I do like the test since we use actual command but not too sure about the way

Reply:
I'm not crazy about this either, but agree with the benefits. Still getting acquainted with common patterns around Golang's Context. There seems to be lack of clarity in the community around testing with this CLI framework. 

An alternative is passing in a factory method, which dictates how the `elReader` is generated. This approach isn't idiomatic go, however. I generally prefer the factory method approach, but I'm coming from Java ideology, and I understand no many people like factories and their verbosity/OO bloat (if used pervasively).

Reply:
I am leaning towards the factory pattern as a resolution to this. Let me know if strongly disagree.

Reply:
I am still not sure about this but we can go ahead with what you think makes sense and revisit it if we have new ways.